dispatch/dispatch_test.go

```diff
@@ -373,7 +374,9 @@ route:
 
 	timeout := func(d time.Duration) time.Duration { return time.Duration(0) }
 	recorder := &recordStage{alerts: make(map[string]map[model.Fingerprint]*types.Alert)}
-	dispatcher := NewDispatcher(alerts, route, recorder, marker, timeout, logger, NewDispatcherMetrics(prometheus.NewRegistry()))
+	lim := limits{groups: 6}
```

Comment:
With the current implementation of the code, this is fine. However, I could see in the future optimizations that would change the way we deal with the limit.

I would encourage you to split this test to avoid regressions in both paths - with and without limits.

Reply:
That's a good idea. I've extracted (and simplified) `TestGroupsWithLimits` test from previous `TestGroups`.